token/program-2022/src/extension/default_account_state/instruction.rs

```diff
@@ -0,0 +1,115 @@
+use {
+    crate::{
+        check_program_account, error::TokenError, instruction::TokenInstruction,
+        state::AccountState,
+    },
+    num_enum::{IntoPrimitive, TryFromPrimitive},
+    solana_program::{
+        instruction::{AccountMeta, Instruction},
+        program_error::ProgramError,
+        pubkey::Pubkey,
+    },
+    std::convert::TryFrom,
+};
+
+/// Default Account State extension instructions
+#[derive(Clone, Copy, Debug, PartialEq, IntoPrimitive, TryFromPrimitive)]
+#[repr(u8)]
+pub enum DefaultAccountStateInstruction {
+    /// Initialize a new mint with the default state for new Accounts.
```

Comment:
🤔 so this is just the base account state right?  That is we can't set default extension state currently.

TBH, I'm slightly worried this is overall more than what we really need, which I think is just the ability to force new accounts to be frozen

Reply:
It could be a bool instead for "init Frozen". Doesn't change the data-size of the extension, but the processing would be slightly simpler (but also less flexible, if there are additional AccountState variants in the future).